src/Promise.php

```diff
@@ -94,9 +94,14 @@ public function catch(callable $onRejected): PromiseInterface
         });
     }
 
+    /**
+     * @param callable(): (void|PromiseInterface<void>) $onFulfilledOrRejected
+     * @return PromiseInterface<T>
+     */
```

Comment:
Tested this locally and this doesn't appear to be needed, as its types would be inherited directly from the `PromiseInterface`:

```suggestion
```

May I ask you do confirm?